lib/submissions_handler.rb

```diff
@@ -81,9 +81,30 @@ def self.process_upload(file, isMapEnabled, mapfile, assignment)
     # Add filters for file types
     accepted_formats = [".py",".java", ".cpp", ".c", ".h", ".scala", ".m", ".ml", ".mli", ".r"]
 
+    # check zip file from Mac
+    zip_from_Mac = false
+    has_entry_same_name_with_upload_file = false
+    upload_file_without_ext = File.basename(upload_file, ".zip") + File::SEPARATOR
+
     # Extract submissions into dir
     Zip::File.open(upload_file) { |zip_file|
       zip_file.each { |f|
+
+      file_entry_names = zip_file.entries.collect {|file| file.name}
+      file_entry_names.each { |file_name|
+        if (file_name.eql?(upload_file_without_ext)) 
+          has_entry_same_name_with_upload_file = true
+        end
+
+        if (file_name.include?("__MACOSX")) 
```

Comment:
Is this directory universally true of such zip files?  Can we include a URL for the documentation?

Reply:
I couldn't find official documentation. Dev forum suggests that other devs also have this issue with zipping files using Mac GUI "Compress files".